### PR TITLE
created settings sidebar

### LINKED ISSUE #417: migrate(react/next): settings:sidebar
Create a sidebar for settings page similar to nuxt application

### PR DESCRIPTION
**Pull request by Ankur1493**
## What does this PR do?

Created the sidebar for settings page, completed the functionality, some pages are not yet created so will give 404, otherwise it's working here are some ss
Screenshot from 2024-05-04 18-31-10
Screenshot from 2024-05-04 18-31-43

Fixes #417 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [x] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [x] Commented on my code in hard-to-understand areas
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [x] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 1]
```diff
@@ -1,11 +1,210 @@
 'use client';
 
-import { Flex } from '@radix-ui/themes';
+import React from 'react';
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:3]: just a few nitpicks,

```suggestion
import { type ReactNode } from 'react';
```

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 2]
```diff
@@ -1,11 +1,210 @@
...
+  Users,
+  UserRoundPlus,
+  HeartHandshake
+} from 'lucide-react';
+
+interface links {
+  label: string;
+  to: string;
+  icon: React.ReactNode;
+}
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:26]:
```suggestion
type NavLinks = {
  label: string;
  to: string;
  icon: ReactNode;
};
```

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 3]
```diff
@@ -1,11 +1,210 @@
...
 
 export default function SettingsSidebar() {
+  const orgShortCode = useGlobalStore((state) => state.currentOrg.shortCode);
+  const pathname = usePathname();
+
+  const personalLinks: links[] = [
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:32]:
```suggestion
  const personalLinks: NavLinks[] = [
```

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 4]
```diff
@@ -1,11 +1,210 @@
...
+      to: `/${orgShortCode}/settings/user/addresses`,
+      icon: <Mail />
+    },
+    {
+      label: 'Security',
+      to: `/${orgShortCode}/settings/user/security`,
+      icon: <Lock />
+    }
+  ];
+  const orgSetupLinks = [
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:49]:
```suggestion
  const orgSetupLinks: NavLinks[] = [
```

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 5]
```diff
@@ -1,11 +1,210 @@
...
+      icon: <Building2 />
+    },
+    {
+      label: 'Billing',
+      to: `/${orgShortCode}/settings/org/setup/billing`,
+      icon: <CreditCard />
+    }
+  ];
+
+  const orgUserLinks: links[] = [
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:62]:
```suggestion
  const orgUserLinks: NavLinks[] = [
```

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx [hunk 6]
```diff
@@ -1,11 +1,210 @@
...
+      to: `/${orgShortCode}/settings/org/users/invites`,
+      icon: <UserRoundPlus />
+    },
+    {
+      label: 'Teams',
+      to: `/${orgShortCode}/settings/org/users/teams`,
+      icon: <HeartHandshake />
+    }
+  ];
+  const orgMailLinks: links[] = [
```
Comment (BlankParticle) [inline apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:79]:
```suggestion
  const orgMailLinks: NavLinks[] = [
```

**Review by BlankParticle — CHANGES_REQUESTED** [review]

File: apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx (hunk 6 above)
Comment (Ankur1493) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:79]: okk, should I do NavLinks or settingLinks?

Comment (BlankParticle) [reply apps/web/app/[orgShortCode]/settings/SettingsSidebar.tsx:79]: anything you like just call it like `CapitalName`

**Review by BlankParticle — DISMISSED** [review]

**Comment by McPizza0** [comment]
Hey @Ankur1493 !
nice work

it seems like its missing the check to see if the user is an admin of the org, and conditionally showing the org settings section based on that

please check the first rows in `apps/web-app/components/settings/sidebar.vue` to see how we currently handle this in nuxt

**Comment by Ankur1493** [comment]
will do it today, was busy with some college work

**Pull request merged**